src/condition.cpp

```diff
@@ -2012,16 +2010,16 @@ bool ConditionManaShield::startCondition(Creature* creature)
 	return false;
 }
 
-void ConditionManaShield::endCondition(Creature* creature)
+void ConditionManaShield::endCondition(std::shared_ptr<Creature> creature)
 {
-	if (Player* player = creature->getPlayer()) {
+	if (auto player = creature->getPlayer()) {
 		player->sendStats();
 	}
 }
 
-void ConditionManaShield::addCondition(Creature* creature, const Condition* addCondition)
+void ConditionManaShield::addCondition(std::shared_ptr<Creature> creature, const Condition* addCondition)
 {
-	if (Player* player = creature->getPlayer()) {
+	if (auto player = creature->getPlayer()) {
```

Comment:
Could this be `const auto`
```suggestion
	if (const auto player = creature->getPlayer()) {
```